lib/rspec/mocks/error_generator.rb

```diff
@@ -291,6 +303,19 @@ def format_received_args(*args)
         args.empty? ? "(no args)" : "(" + received_arg_list(*args) + ")"
       end
 
+      def format_unexpected_with_args(*args)
+        with_args = Hash[Array(args.first).group_by { |x| x }.map { |k, v| [k, v.count] }]
```

Comment:
I was initially trying to process all the responses through a shared method, and that was just a way to ensure it would always call group_by on a Array. Left it in after for better fault tolerance.